pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -83,6 +91,46 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	}
 }
 
+// tagImages tags images from the deployment
+func (e *HookExecutor) tagImages(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationController, label string) error {
+	config, err := deployutil.DecodeDeploymentConfig(deployment, e.decoder)
+	if err != nil {
+		return err
+	}
+
+	var errs []error
+	for _, action := range hook.TagImages {
+		image, err := deployapi.TemplateImageForContainer(&deployment.Spec.Template.Spec, deployapi.DeploymentConfigHasTrigger(config), action.ContainerName)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("unable to find image for container %q: %v", action.ContainerName, err))
+			continue
+		}
+		value := image.Image
+		namespace := action.To.Namespace
+		if len(namespace) == 0 {
+			namespace = deployment.Namespace
+		}
+		if _, err := e.tags.ImageStreamTags(namespace).Update(&imageapi.ImageStreamTag{
+			ObjectMeta: kapi.ObjectMeta{
+				Name:      action.To.Name,
```

Comment:
Yeah, I'm pretty convinced that `hook.TagImages.To.Kind` should be `ImageStreamTag`.  That's what we're updating here.